src/jit/emitxarch.cpp

```diff
@@ -12094,7 +12094,7 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
                 if (IsDstDstSrcAVXInstruction(ins))
```

Comment:
Are there no cases where we use this format for an `IsDstSrcSrcAvxInstruction()`? I'm having trouble imagining how that would make sense for these formats. Would an assert make sense?

Reply:
We should actually not have this check (or a `IsDstSrcSrcAvxInstruction`) check at all (for these formats).

Those two checks are meant for the case where we call a 2 operand encoding (such as `IF_RRW_MRD_CNS`) and the third operand is implicitly determined.

This particular format is used when the emitter was explicitly given the register for the 2nd operand, so there is no checks needed.

Reply:
Right, so perhaps we should simply remove it. I think that might be clearer. And perhaps there could be an assert that we're using AVX encodings?

Reply:
Was just working on doing that 😄

Reply:
@CarolEidt, should be fixed now. Also updated `IF_RWR_RRD_MRD` and `IF_RWR_RRD_SRD`, since they were incorrect as well.